Add PWA page with explanation (#16697)
decidim/decidim: docs/modules/services/pages/pwa.adoc

= Progressive Web App

Decidim supports Progressive Web App features to make the platform feel closer to a mobile app.

> For operators, the main value is an installable experience, WebPush notifications, and the option to package the same web app for app stores.

Author: ```suggestion
For operators, the main value is a web app experience with optional installation, WebPush notifications, and the option to package the same web app for app stores.
```

== What it includes

> - Installable web app experience through the browser

Author: ```suggestion
- App-like experience in the browser with optional installation
```

== WebPush setup

To use push notifications, first configure the VAPID keys in your application environment.
See xref:configure:environment_variables.adoc[Environment Variables] for the supported variables.

Generate a key pair with:

[source,bash]
----
bin/rails decidim:pwa:generate_vapid_keys
----

The command prints the public and private keys.
Set them as `VAPID_PUBLIC_KEY` and `VAPID_PRIVATE_KEY`, then restart the application.

To test it, log in with any participant account and open the notification settings in the profile.
There should be an option to enable WebPush notifications.
After enabling it, new notifications should also be received natively according to the platform being used.

== Packaging with PWABuilder

Use https://www.pwabuilder.com/[PWABuilder] to package the Decidim PWA for app stores.

1. Make sure the site exposes a valid web app manifest and service worker.
2. Open https://www.pwabuilder.com/[PWABuilder] and enter the Decidim URL.
3. Review the detected PWA features and fix any issues it reports.
4. Generate the package you need:
   - Android, for Google Play workflows
   - iOS, for Apple distribution workflows

>    - Windows, for Microsoft Store or sideloading

Author: ```suggestion
   - Windows, for Microsoft Store or for installation outside of the Store
```
